wallet/datastore.go

```diff
@@ -431,47 +357,710 @@ func (pg *Postgres) IncreaseLinkingLimit(ctx context.Context, providerLinkingID
+	} else if !ok {
+		// this wallet has not yet been migrated to the CustodianLink model
+		// perform the migration which does 1-3 from above
+		if err := pg.MigrateCustodianLink(ctx, walletID); err != nil {
+			return fmt.Errorf("failed to migrate custodian linkage: %w", err)
+		}
+	}
+	// 4.) Continue processing this method
+
+	// sql query to perform
+	stmt := `
+		update wallet_custodian as wc
+		set disconnected_at=now()
+		from wallets as w
+		where 
+			w.wallet_custodian_id=wc.id
+			and w.id=$1
+	`
+	// perform query to set disconnected time on wallet custodian
+	if r, err := tx.ExecContext(
+		ctx,
+		stmt,
+		walletID,
+	); err != nil {
+		sublogger.Error().Err(err).Msg("failed to update wallet_custodian_id for wallet")
+		return err
+	} else if r != nil {
+		count, _ := r.RowsAffected()
+		if count < 1 {
+			sublogger.Error().Msg("at least one record should be updated for disconnecting a verified wallet")
+			return errors.New("should have updated at least one wallet with disconnected custodial")
+		}
+	}
+
+	// sql query to perform unlinking
+	stmt = `
+		update wallets set wallet_custodian_id=null where id=$1
+	`
+	// perform query
+	if r, err := tx.ExecContext(
+		ctx,
+		stmt,
+		walletID,
+	); err != nil {
+		sublogger.Error().Err(err).Msg("failed to update wallet_custodian_id for wallet")
+		return err
+	} else if r != nil {
+		count, _ := r.RowsAffected()
+		if count < 1 {
+			sublogger.Error().Msg("at least one record should be updated for disconnecting a verified wallet")
+			return errors.New("should have updated at least one wallet with disconnected custodial")
+		}
+	}
+
+	// commit
+	err = tx.Commit()
 	if err != nil {
-		return errorutils.Wrap(err, "error looking up max linkings for wallet")
+		sublogger.Error().Err(err).Msg("failed to commit transaction")
+		return fmt.Errorf("failed to commit transaction: %w", err)
 	}
 
-	usedLinkings, err := txGetUsedLinkingSlots(ctx, tx, providerLinkingID.String())
+	// done
+	return nil
+}
+
+// InsertCustodianLink - create a record of a custodian wallet
+func (pg *Postgres) InsertCustodianLink(ctx context.Context, cl *CustodianLink) error {
+	var err error
+	// create a sublogger
+	sublogger := logger(ctx).With().
+		Str("wallet_id", cl.GetWalletIDString()).
+		Str("custodian", cl.Custodian).
+		Str("linking_id", cl.GetLinkingIDString()).
+		Logger()
+
+	sublogger.Debug().
+		Msg("creating linking of wallet custodian")
+
+	// get tx
+	tx, noContextTx := ctx.Value(appctx.DatabaseTransactionCTXKey).(*sqlx.Tx)
+	if !noContextTx {
+		tx, err = createTx(ctx, pg)
+		if err != nil || tx == nil {
+			sublogger.Error().Err(err).
+				Msg("error creating tx for wallet linking")
+			return fmt.Errorf("failed to create tx for wallet linking: %w", err)
+		}
+		// add tx to ctx for future
+		defer pg.RollbackTx(tx)
+	}
+
+	// TODO: check here that prior provider linking ids match this new custodian link entry for the given provider
```

Comment:
should we delete this?

Reply:
resolved: https://github.com/brave-intl/bat-go/pull/793/commits/ca66e903529564cb9ad4e5af367a97fda7fa4f44